File: M09-PSP/UF1/POO/dam2-m9/intro_poo/activitats/activitat_u/java/activitat_u/src/conceptes/classes/Cap.java
package conceptes.classes;

public class Cap extends Treballador {
    private double incentiu;

    public Cap(String nom, double sou, int any, int mes, int dia) {
        super(nom, sou, any, mes, dia);
    }

    public void setIncentiu(double incentiu) {
        this.incentiu = incentiu;
    }

    public double getSou() {
        double souCap = super.getSou();
        return souCap + incentiu;

    }
}

File: M09-PSP/UF2/Teoria/dam2-m9/intro_poo/activitats/activitat_u/python/py_activitat_01/conceptes/cap.py
from conceptes.treballador import Treballador


class Cap(Treballador):
    def __init__(self, nom, sou, incentiu):
        self._incentiu = incentiu
        super().__init__(nom, sou)

    @property
    def incentiu(self):
        return self._incentiu

    @incentiu.setter
    def incentiu(self, incentiu):
        self._incentiu = incentiu

    @property
    def sou(self):
        return super().sou + self.incentiu

    def __str__(self) -> str:
        return super().__str__() + f'\nIncentiu {self.incentiu}'
